fipy/tools/numerix.py

```diff
@@ -1007,6 +1007,48 @@ def in1d(ar1, ar2, assume_unique=False):
         else:
             return flag[indx][rev_idx]
 
+def lists_to_masked(x):
+    """Convert list of lists of different length to Numpy masked array
+
+    See https://stackoverflow.com/questions/38619143/convert-python-sequence-to-numpy-array-filling-missing-values
+
+    >>> print(lists_to_masked([[0], [0, 1], [0, 1, 2]]))
+    [[0 0 0]
+     [-- 1 1]
+     [-- -- 2]]
+
+    After https://stackoverflow.com/a/59686318/2019542
+    """
+    import itertools
+
+    return MA.masked_values(list(itertools.zip_longest(*x, fillvalue=-1)), -1)
+
+
+def invert_indices(arr, axis=-1):
```

Comment:
Should this actually be in numpy proper?

Reply:
Which? `lists_to_masked` or `invert_indices`? Or both?

Reply:
It would probably need to be in scipy due to the dependence on `scipy.sparse.coo_matrix`

Reply:
Oh yes, good point. Both could be useful I think. Not something to worry about now though.